### PR TITLE
feat: Implement Start Recording HTTP endpoint

### LINKED ISSUE #15: Implement Record Start Endpoint
Per ADR: https://docs.edgexfoundry.org/3.0/design/adr/application/0025-Record-and-Replay/#post

Create initial Data Manager Interface and Mock with just the `StartRecording` API

Unit test endpoint using mock Data Manager

### PR DESCRIPTION
**Pull request by lenny-intel**
closes #15

**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/app-record-replay/blob/main/.github/CONTRIBUTING.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I am not introducing a new dependency (add notes below if you are)
- [x] I have added unit tests for the new feature or bug fix (if not, why?)
- [ ] I have fully tested (add details below) this the new feature or bug fix (if not, why?) **N/A unit tests suffice**
- [ ] I have opened a PR for the related docs change (if not, why?) **N/A not yet**
  <link to docs PR>

### CODE REVIEW

File: internal/controller/controller.go [hunk 1]
```diff
@@ -79,10 +84,29 @@ func (c *httpController) AddRoutes() error {
...
+	if err := json.NewDecoder(request.Body).Decode(&startRequest); err != nil {
+		writer.WriteHeader(http.StatusBadRequest)
+		_, _ = writer.Write([]byte(fmt.Sprintf("%s: %v", failedRequestJSON, err)))
+		return
+	}
+
+	if startRequest.Duration == 0 && startRequest.EventLimit == 0 {
+		writer.WriteHeader(http.StatusBadRequest)
+		_, _ = writer.Write([]byte(failedRecordRequestValidate))
+		return
```
Comment (jim-wang-intel) [inline internal/controller/controller.go:100]: should we also validate the value must be non-negative .. so some meaningful value to have?

Comment (lenny-intel) [reply internal/controller/controller.go:100]: Good catch. Will add

Comment (lenny-intel) [reply internal/controller/controller.go:102]: added

**Review by jim-wang-intel — APPROVED** [review]
LGTM

**Pull request merged**